source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -442,6 +446,35 @@ ShaderGraphPtr ShaderGraph::create(const ShaderGraph* parent, const NodeGraph& n
     return graph;
 }
 
+void ShaderGraph::propagateAddedInputs(ShaderNode& node, const StringVec& addedInputs)
+{
+    if (!_propagateAddedInputs) {
+        return;
+    }
+
+    for (auto const& name: addedInputs) {
+        auto* nodeInput = node.getInput(name);
+        if (nodeInput) {
+            auto* inputSocket = getInputSocket(name);
+            if (!inputSocket) {
+                // Only add an input socket once. This means that a NodeGraph
+                // with three embedded image nodes will only get a single
+                // pair of uv_scale and uv_offset inputs shared between
+                // all three nodes.
+                // TODO: Have as many uv_scale inputs as there are embedded image
+                //       nodes. Requires a scheme to make their names unique. Bonus
+                //       points if that name can be related to the filename input.
+                // NOTE: This is enough to have UDIMs working with ND_gltf_colorimage
+                //       and ND_UsdUVTexture, so we stop here at this time.
+                inputSocket = addInputSocket(name, nodeInput->getType());
```

Comment:
Original comment from @kwokcb:
> Hi @JGamache-autodesk,
> I don't think this is enough and is inconsistent with how other sockets are "published" using <node name>_ <input > name> . This would also resolve the issue that you should be able to support different UDIM scale/offsets for > different image inputs -- i.e. there is no guarantee there is only one set of UDIM information.

Reply by @JGamache-autodesk:
> We should definitely split the issue, especially since this PR only attempts to fix single image wrappers like those found for glTf, USD, ADSK, Arnold, etc. Image nodes wrapped in simple NodeGraph implementations.
> The second part where we support multi-image nodes with multiple UDIM scales should be indeed discussed and implemented correctly, but maybe should not be a showstopper for first fixing ND_UsdUVTexture and ND_gltf_colorimage.

Reply:
@kwokcb please advise: Is it a case where I should rework the comment, or is it a case where a fix should only be submitted if it also covers the edge case mentioned in the comment?